packages/effect/src/platform/Terminal.ts

```diff
@@ -0,0 +1,95 @@
+/**
+ * @since 4.0.0
+ */
+
+import { Effect, Scope, ServiceMap } from "../index.ts"
+import type { Dequeue } from "../Queue.ts"
+import type { PlatformError } from "./index.ts"
+import type * as Option from "../data/Option.ts"
+import * as Data from "../data/Data.ts"
+
+/**
+ * A `Terminal` represents a command-line interface which can read input from a
+ * user and display messages to a user.
+ *
+ * @since 4.0.0
+ * @category models
```

Comment:
Let's improve the categories for consistency so that things get nicely grouped in the output API documentation.

- `models` / `model` -> `models`
- `tag` -> `services`

We also may want to consider adding a `TypeId` to the `Terminal` interface like to do with `FileSystem`, so that we can also add a type guard (i.e. `isTerminal`). This would also necessitate us adding a `make` constructor to the model that automatically adds the `TypeId` (so we can keep the `TypeId` internal). See the `FileSystem` module for what I mean.